core/distributed/matrix.cpp

```diff
@@ -140,8 +141,10 @@ void Matrix<ValueType, LocalIndexType, GlobalIndexType>::move_to(
 template <typename ValueType, typename LocalIndexType, typename GlobalIndexType>
 void Matrix<ValueType, LocalIndexType, GlobalIndexType>::read_distributed(
     const device_matrix_data<value_type, global_index_type>& data,
-    const Partition<local_index_type, global_index_type>* row_partition,
-    const Partition<local_index_type, global_index_type>* col_partition)
+    const gko::distributed::Partition<local_index_type, global_index_type>*
+        row_partition,
+    const gko::distributed::Partition<local_index_type, global_index_type>*
+        col_partition)
```

Comment:
I didn't move `Partition´ because it's already in develop, but we can still move it into the experimental namespace. I don't have a strong opinion on that.